src/coreclr/jit/fgopt.cpp

```diff
@@ -1882,16 +1881,11 @@ void Compiler::fgCompactBlocks(BasicBlock* block, BasicBlock* bNext)
             break;
     }
 
-    // Add the LOOP_ALIGN flag
     if (bNext->isLoopAlign())
     {
-        // Only if the new block is jump target or has label
```

Comment:
I added this condition because I did see cases where we were wrongly adding the `BBF_LOOP_ALIGN` flag on blocks that no longer constitute a loop. Perhaps, the failures that you are seeing in CI run is related to that. You might also want to double check the `loopsAligned` stats before vs. after your change because after relaxing this condition, we might be aligning too many loops (some of them are not valid candidates anymore).

Reply:
The problem is that the number of loop aligned changes with my changes to the set of labels applied, so any before/after will mostly (I think) not be related to this case.

We don't have labels at this point in compilation, so we can't do the check here that was previously done.

I don't quite understand why this check was there. Say we're compacting BB1 and BB2. And BB2 is marked BBF_LOOP_ALIGN, meaning it's the top of a loop. Wouldn't the compacted BB1+BB2 also be the top of a loop? Is this a case where your recent change to `fgUpdateLoopsAfterCompacting` would fix any problem that you were seeing before?

Reply:
If I recall correctly, it was the case for which you are seeing the CI failures (I may be wrong). In the original code, you might add an assert around that code and you should find cases where the hypothesis of compacting loopTop is not true.

```
    if (bNext->isLoopAlign())
    {
        // Only if the new block is jump target or has label
        if (((block->bbFlags & BBF_JMP_TARGET) != 0) || ((block->bbFlags & BBF_HAS_LABEL) != 0))
        {
            block->bbFlags |= BBF_LOOP_ALIGN;
            JITDUMP("Propagating LOOP_ALIGN flag from " FMT_BB " to " FMT_BB " during compacting.\n", bNext->bbNum,
                    block->bbNum);
        }
        else
       {
         assert(!"This is the case");
       }
    }

Reply:
No SPMI diffs with that assert. I triggered a PR with the assert to see if it occurs elsewhere: https://github.com/dotnet/runtime/pull/51290